py/kubeflow/testing/run_e2e_workflow.py

```diff
@@ -454,13 +483,15 @@ def run(args, file_handler): # pylint: disable=too-many-statements,too-many-bran
       ui_urls[workflow_name] = ui_url
       logging.info("URL for workflow: %s", ui_url)
 
-  ui_urls.update(tekton_runner.run(
-      tekton_client.ClusterInfo(args.project,
-                                TEKTON_CLUSTER_ZONE,
-                                TEKTON_CLUSTER_NAME),
-      tekton_client.ClusterInfo(args.project, args.zone, args.cluster)))
-  # We delay creating started.json until we know the Argo workflow URLs
-  create_started_file(args.bucket, ui_urls)
+  if not args.cloud_provider:
```

Comment:
Yes. Argo and Tekton both should be supported.

Reply:
Yes and I think for now we use ksonnet as a start-point. It can help us leverage upstream existing tests.

Meanwhile we'll add support for Tekton in AWS.